tests/python_package_test/test_engine.py

```diff
@@ -29,6 +30,34 @@ def dummy_obj(preds, train_data):
     return np.ones(preds.shape), np.ones(preds.shape)
 
 
+def mse_obj(y_pred, dtrain):
+    y_true = dtrain.get_label()
+    grad = (y_pred - y_true)
+    hess = np.ones(len(grad))
+    return grad, hess
+
+
+def logloss_obj(preds, train_data):
+    """Taken from https://maxhalford.github.io/blog/lightgbm-focal-loss/"""
+    y = train_data.get_label()
+    p = special.expit(preds)
+    grad = p - y
+    hess = p * (1 - p)
+    return grad, hess
+
+
+def logloss_metric(preds, train_data):
```

Comment:
as mentioned in https://github.com/microsoft/LightGBM/pull/5052#pullrequestreview-913851568, please remove anything related to custom *metric* from this PR, to keep it focused only on custom objective functions

Reply:
copy that